app/models/product_accessory.rb

```diff
@@ -9,10 +13,17 @@ class ProductAccessory < ActiveRecord::Base
   ## validations
   validates :product, presence: true
   validates :accessory, presence: true
-  validates :scaling_type, presence: true, inclusion: SCALING_TYPES
+  validates :scaling_type, presence: true
+  validate :scaling_type_matches_product?
```

Comment:
I would probably lean away from using the `?` in the method name since it's not really a predicate.